### PR TITLE
WIP: Sync maintainer scripts etc with latest downstream 10.11.5 in Debian

### PR DESCRIPTION
**Pull request by ottok**
## Description

Sync downstream packaging improvements in Debian to upstream. See commit messages for details.

*When merging, keep all commits separate so their commit history remains legible.*

## How can this PR be tested?

Best tested by verifying there are no regressions on Buildbot nor Salsa-CI.

## Basing the PR against the correct MariaDB version

This is based on 10.11 because source was Debian 10.11. I have not checked what upstream branch is the truly most upstream branch to land these changes on - potentially 10.7 as that is first branch with fuly versionless 'mariadb-server' packages.

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.0/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [no] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -30,8 +30,7 @@ Build-Depends: bison,
                liblz4-dev,
                liblzma-dev,
                liblzo2-dev,
-               libncurses5-dev (>= 5.0-6~),
-               libncurses5-dev:native (>= 5.0-6~),
+               libncurses-dev,
```
Comment (illuusio) [inline debian/control:33]: This one should be handled in #2794 because it adds needed packages back in older distros

**Comment by illuusio** [comment]
As this seems good to me what what is reason to use `command -v` over the `which`?

**Comment by grooverdan** [comment]
> As this seems good to me what what is reason to use `command -v` over the `which`?

I assume command -v is a shell internal as opposed to which.

Apart from removing the curses changes that are already merged, is this ready?

**Comment by illuusio** [comment]
I've rebased this to current 10.11 and run on Salsa-CI https://salsa.debian.org/illuusio/mariadb-server/-/commit/aa9a6bca65ef5ae115a2cf6a76b1332ec7e9fb80

**Comment by ottok** [comment]
I am closing this PR as I don't have time to work on it, doing the primary maintenance in Debian is quite a lot by itself already and I need to focus on it to sort out 64-bit time issues and to make sure MariaDB has a good version for Ubuntu 24.04.

Tuukka: feel free to copy as much of my commits from Debian back upstream. All the code is GPLv2 so there are no barriers for you to simply copy as much as you want.

**Pull request closed without merging**